src/ext/java/org/opentripplanner/ext/flex/template/ClosestTrip.java

```diff
@@ -0,0 +1,134 @@
+package org.opentripplanner.ext.flex.template;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.opentripplanner.ext.flex.trip.FlexTrip;
+import org.opentripplanner.framework.lang.IntUtils;
+import org.opentripplanner.routing.graphfinder.NearbyStop;
+import org.opentripplanner.transit.model.timetable.booking.RoutingBookingInfo;
+
+/**
+ * The combination of the closest stop, trip and trip active date.
+ */
+record ClosestTrip(
+  NearbyStop nearbyStop,
+  FlexTrip<?, ?> flexTrip,
+  int stopPos,
+  FlexServiceDate activeDate
+) {
+  ClosestTrip(
+    NearbyStop nearbyStop,
+    FlexTrip<?, ?> flexTrip,
+    int stopPos,
+    FlexServiceDate activeDate
+  ) {
+    this.nearbyStop = Objects.requireNonNull(nearbyStop);
+    this.flexTrip = Objects.requireNonNull(flexTrip);
+    this.stopPos = IntUtils.requireNotNegative(stopPos, "stopPos");
+    this.activeDate = activeDate;
```

Comment:
Is activeDate nullable or can you do `Objects.requireNonNull`?

Reply:
If it's nullable, can you add an annotation.

Reply:
Ah, i see the method further down...

Reply:
This little trick is ok since it is an internal package-local class - it would not be ok if it was public. I decided not to make 2 classes, one with and one without the `activeDate`. Because it is 2-steps to create the instance, the date can not be verified as nonnull in the constructor. This is possible something I would like to change when taking a look at the design later. There is a bug here as well (I did not fix it because it might have a negative effect on performance). The bug is because only the best "nearest" path is kept, but if you are between to stops it might be better to walk to the stop witch is 2 more meters away if it means that the FLEX service will not drive back that same way as you walked. In extreme cases this bug will cause you to get no result at all. I tested it, and was able to reproduce this.